app/controllers/saml_idp_controller.rb

```diff
@@ -133,9 +133,25 @@ def capture_analytics
       matching_cert_serial: saml_request.service_provider.matching_cert&.serial&.to_s,
       requested_nameid_format: saml_request.name_id_format,
     )
+
+    if result.success? && saml_request.signed?
+      # Logging to indicate if a validation bug fix will create a potentially breaking change
+      analytics_payload[:encryption_cert_matches_matching_cert] =
+        encryption_cert_matches_matching_cert?
+    end
```

Comment:
would it be valuable to have this key regardless of if the result is successful? would it be useful to add a `request_signed` attribute also?

Reply:
1. i'm not sure! this logging is intended to determine which partners would be affected by successful responses that are returning responses encrypted with the wrong certificate. it's unclear to me if there is value in this field if there's no encrypted SAML response, what do you think?

2. there is a `request_signed` attribute, but i thought it would be confusing to return this value if it wasn't signed, since it will always be `false` (`saml_request.matching_cert` is `nil` if the certificate isn't signed)

Reply:
I feel like having the key available sometimes would be trickier to understand? but it would make it easier to search for stuff where this applies 🤷 . I'm not sure anymore

Reply:
i'm just worried about people drawing conclusions from false negatives is the query isn't specific eenough. do you wanna meet today to talk through the use case briefly? i may be overly cautious.

Reply:
could return `nil` in scenarios where it doesn't matter (ie, not successful and not signed) -- although values that can evaluate to true/false/nil have bitten me in the past, so i'm always a bit wary of those.

Reply:
Unconditionally setting the `encryption_cert_matches_matching_cert` field could make sense because we also capture the request validation state (i.e., `result.success`). However, any log queries would have also have to include a filter of `properties.event_properties.success = true AND properties.event_properties.request_signed = true` to only look at the events relevant to the SAML validation fix.

@Sgtpluck - is this field a permanent addition or only until the validation fix is verified to be working as expected and not breaking anything for partners?

Reply:
i imagined it as a temporary addition, as i think it has limited utility long-term, although we could discover that's not the case.

and yes, we can filter in the queries. but my concern was any confusion that may arise if someone didn't include the filters and drew incorrect conclusions from the field, since the person doing the query has to know that context.

Reply:
(i think we're saying the same thing when i reread!)

Reply:
We chatted about it and decided to keep as-is (since it is intended to be temporary)